int-set/src/lib.rs

```diff
@@ -152,19 +163,29 @@ impl<T: Domain<T>> IntSet<T> {
     /// Returns an iterator over all disjoint ranges of values within the set in sorted ascending order.
     pub fn iter_ranges(&self) -> impl Iterator<Item = RangeInclusive<T>> + '_ {
         let u32_iter = match &self.0 {
-            Membership::Inclusive(s) => RangeIter::Inclusive {
-                ranges: s.iter_ranges(),
-            },
+            Membership::Inclusive(s) => {
```

Comment:
what's the relationship between these changes and the nominal goals of the patch?

Reply:
This is a fix for incorrect behaviour found by the fuzzer additions. In this case the iteration over ranges discontinous was incorrect: ranges that were actually adjacent (but not in u32 space) were being returned as disjoint.